etc/microshift/manifests/opentelemetry-gateway.yaml

```diff
@@ -0,0 +1,82 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: opentelemetry-gateway-config
+  namespace: observability
+data:
+  # TODO: add k8s/microshift specific configuration
+  config.yaml: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
+    processors:
+      memory_limiter:
+        check_interval: 1s
+        limit_percentage: 75
+        spike_limit_percentage: 15
+      batch:
+        send_batch_size: 10000
+        timeout: 10s
+    exporters:
+      otlp/host:
+        endpoint: localhost:4317
+        tls:
+          insecure: true
+          insecure_skip_verify: true
+    service:
+      pipelines:
+        traces:
+          receivers: [otlp]
+          processors: []
+          exporters: [otlp/host]
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: opentelemetry-gateway
+  namespace: observability
+  labels:
+    app: opentelemetry-gateway
+spec:
+  # NOTE: Access the container on the host network.
+  # TODO: Does this expose ports on the hostnetwork too and could lead
+  # to conflicts with the default otlp ports?
+  hostNetwork: true
+  containers:
+  - name: opentelemetry-collector
+    image: registry.redhat.io/rhosdt/opentelemetry-collector-rhel8
+    command: ["/usr/bin/opentelemetry-collector", "--config=/etc/opentelemetry/config.yaml"]
+    volumeMounts:
+    - name: config-volume
+      mountPath: /etc/opentelemetry
+  volumes:
+  # TODO: Consider hostPath as alternative
```

Comment:
I like more the configmap approach to keep everything in one place

Reply:
What I also had in mind was a pvc.
```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: opentelemetry-collector-config-pv
spec:
  capacity:
    storage: 5Mi
  accessModes:
    - ReadOnlyMany
  hostPath:
    path: /etc/opentelemetry-collector-rhde-config/
    type: DirectoryOrCreate
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: opentelemetry-collector-config-pvc
spec:
  accessModes:
    - ReadOnlyMany
  resources:
    requests:
      storage: 5Mi
  selector:
    matchLabels:
      app: opentelemetry-collector
```
But I am fine with the configmap. Thats easy :)